src/app/core/view/dynamic-components/dynamic-component.decorator.ts

```diff
@@ -0,0 +1,33 @@
+import { OnInitDynamicComponent } from "./on-init-dynamic-component.interface";
+import { viewRegistry } from "../../registry/dynamic-registry";
+
+/**
+ * Decorator to annotate a class that serves as dynamic component
+ * A dynamic component can be referenced from the config.
+ * The name to reference the component from is the name of the class without the
+ * `Component`, or one of the registered aliases
+ * @example
+ * ```
+ * @DynamicComponent()
+ * @Component(...)
+ * class DoSomethingComponent {
+ *   // Component definition
+ * }
+ *
+ * // Later in some config:
+ *
+ * {
+ *   view: "DoSomething"
+ * }
+ * ```
+ * @param aliases Aliases that the component is also known under
+ * @constructor
+ */
+export function DynamicComponent<
+  Constructor extends new (...args: any[]) => OnInitDynamicComponent
```

Comment:
We could generalize the `EntityConstructor` type towards a general `Constructor` that could be used here like `Constructor<OnInitDynamicComponent>` to make this more readable.
Then the `EntityConstructor` could only be the extension with `Constructor<T extends Entity = Entity> & typeof Entity`